js/ui/components/clubhouse.js

```diff
@@ -200,6 +206,62 @@ var ClubhouseNotificationBanner = new Lang.Class({
         }
     },
 
+    _rearrange_elements: function() {
+        let contentBox = this._bodyStack.get_parent();
+        contentBox.add_style_class_name('clubhouse-content-box');
+        let hbox = contentBox.get_parent();
+        let vbox = hbox.get_parent();
+
+        let wrapBin = new St.Bin({ x_fill: true,
+                                   y_fill: true,
+                                   x_align: St.Align.END,
+                                   y_align: St.Align.END,
+                                   x_expand: true,
+                                   y_expand: true });
+        hbox.add_child(wrapBin);
+
+        // A Clutter.BinLayout is used to rearrange the notification
+        // elements in layers:
+        let wrapWidget = new St.Widget({ layout_manager: new Clutter.BinLayout(),
+                                         x_align: Clutter.ActorAlign.FILL,
+                                         y_align: Clutter.ActorAlign.END });
+        wrapBin.set_child(wrapWidget);
```

Comment:
This is quite a bit of changes. I am curious to know if you could tweak the original content box instead of having a different hierarchy, but it's still OK if not. Perhaps you can add a comment as to why this reparenting it necessary here.

Reply:
@joaquimrocha had the same concern. In fact initially I started tweaking the current layout, which is basically a horizontal box inside a vertical box, and overriding the actor's positions. But I couldn't get it to work and even worse looked too hacky. This way is more elegant IMO because it's the layout as I would do it from scratch, using a `Clutter.BinLayout` to layout the actors in layers, arranging them with their expand/align properties and with CSS.